lua/null-ls/loop.lua

```diff
@@ -203,13 +203,37 @@ M.timer = function(timeout, interval, should_start, callback)
     }
 end
 
-M.temp_file = function(content, extension)
-    local tmp_dir = u.path.is_windows and vim.fn.getenv("TEMP") or "/tmp"
+local tmp_dir
+local get_temp_dir = function()
+    if tmp_dir then
+        return tmp_dir
+    end
+
+    if u.path.is_windows then
+        tmp_dir = vim.fn.getenv("TEMP")
+        return tmp_dir
+    end
+
+    local xdg_runtime_dir = vim.fn.getenv("XDG_RUNTIME_DIR")
```

Comment:
```suggestion
    local xdg_runtime_dir = vim.env["XDG_RUNTIME_DIR"]
```

While testing this, I found that `vim.fn.getenv` actually returns `vim.NIL`. If we use the `vim.env` accessor, we can avoid this issue.

Edit: also, this might make testing a bit simpler, since we can just set and clear the env vars. We can make the same change to the Windows `getenv` call for simplicity.